src/jit/lsraxarch.cpp

```diff
@@ -146,99 +204,104 @@ void LinearScan::BuildNode(GenTree* tree)
         case GT_COMMA:
         case GT_QMARK:
         case GT_COLON:
-            info->srcCount = 0;
-            assert(info->dstCount == 0);
+            srcCount = 0;
+            assert(dstCount == 0);
             unreached();
             break;
 
         case GT_RETURN:
-            BuildReturn(tree);
+            srcCount = BuildReturn(tree);
+            killMask = getKillSetForReturn();
+            BuildDefsWithKills(tree, 0, RBM_NONE, killMask);
             break;
 
         case GT_RETFILT:
-            assert(info->dstCount == 0);
+            assert(dstCount == 0);
             if (tree->TypeGet() == TYP_VOID)
             {
-                info->srcCount = 0;
+                srcCount = 0;
             }
             else
             {
                 assert(tree->TypeGet() == TYP_INT);
-
-                info->srcCount = 1;
-
-                info->setSrcCandidates(this, RBM_INTRET);
-                LocationInfoListNode* locationInfo = getLocationInfo(tree->gtOp.gtOp1);
-                locationInfo->info.setSrcCandidates(this, RBM_INTRET);
-                useList.Append(locationInfo);
+                srcCount = 1;
+                BuildUse(tree->gtOp.gtOp1, RBM_INTRET);
             }
             break;
 
         // A GT_NOP is either a passthrough (if it is void, or if it has
         // a child), but must be considered to produce a dummy value if it
         // has a type but no child
         case GT_NOP:
-            info->srcCount = 0;
+            srcCount = 0;
             assert((tree->gtOp.gtOp1 == nullptr) || tree->isContained());
             if (tree->TypeGet() != TYP_VOID && tree->gtOp.gtOp1 == nullptr)
             {
-                assert(info->dstCount == 1);
+                assert(dstCount == 1);
+                BuildUse(tree->gtOp.gtOp1);
+                BuildDef(tree);
             }
             else
             {
-                assert(info->dstCount == 0);
+                assert(dstCount == 0);
             }
             break;
 
         case GT_JTRUE:
         {
-            info->srcCount = 0;
-            assert(info->dstCount == 0);
+            srcCount = 0;
+            assert(dstCount == 0);
             GenTree* cmp = tree->gtGetOp1();
             assert(!cmp->IsValue());
         }
         break;
 
         case GT_JCC:
-            info->srcCount = 0;
-            assert(info->dstCount == 0);
+            srcCount = 0;
+            assert(dstCount == 0);
             break;
 
         case GT_SETCC:
-            info->srcCount = 0;
-            assert(info->dstCount == 1);
+            srcCount = 0;
+            assert(dstCount == 1);
+            dstCandidates = RBM_NONE;
 #ifdef _TARGET_X86_
-            info->setDstCandidates(this, RBM_BYTE_REGS);
+            dstCandidates = RBM_BYTE_REGS;
 #endif // _TARGET_X86_
+            BuildDef(tree, dstCandidates);
```

Comment:
Or just pass `RBM_BYTE_REGS` directly? It's `RBM_ALLINT` on x64 so it should do the right thing.